### PR TITLE
Define optical MaterialView and rename MaterialParams

### PR DESCRIPTION
**Pull request by sethrj**
This renames `MaterialProperties` to `MaterialParams` and defines a `MaterialView` class. As discussed on slack, it also now validates that the refractive index is defined in all the optical materials.

### CODE REVIEW

File: src/celeritas/optical/CerenkovDndxCalculator.hh [hunk 1]
```diff
@@ -52,94 +52,84 @@ namespace optical
...
                            units::ElementaryCharge charge);
 
     // Calculate the mean number of Cerenkov photons produced per unit length
     inline CELER_FUNCTION real_type operator()(units::LightSpeed beta);
 
   private:
-    NativeCRef<MaterialPropertyData> const& properties_;
-    NativeCRef<CerenkovData> const& shared_;
-    OpticalMaterialId material_;
+    // Calcaulte refractive index [MeV -> unitless]
```
Comment (amandalund) [inline src/celeritas/optical/CerenkovDndxCalculator.hh]:
```suggestion
    // Calculate refractive index [MeV -> unitless]
```

**Review by amandalund — APPROVED** [review]
Looks good @sethrj!

File: test/celeritas/optical/Cerenkov.test.cc [hunk 2]
```diff
@@ -35,10 +35,12 @@
...
 namespace celeritas
 {
+namespace optical
+{
 namespace test
 {
-using namespace celeritas::optical;
-
+//---------------------------------------------------------------------------//
+//
```
Comment (stognini) [inline test/celeritas/optical/Cerenkov.test.cc:43]: Nitpick: Missing comment or extraneous `//`?

Comment (sethrj) [reply test/celeritas/optical/Cerenkov.test.cc:43]: I'll pull that in on the next PR (which actually tackles cerenkov issue)

**Review by stognini — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**